custom-remote-user-store-manager/agent/src/main/java/org/wso2/carbon/remote/user/store/agent/resource/UserManagement.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.remote.user.store.agent.resource;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.remote.user.store.agent.constant.CommonConstants;
+import org.wso2.carbon.remote.user.store.agent.exception.RemoteUserStoreException;
+import org.wso2.carbon.remote.user.store.agent.manager.common.UserStoreManager;
+import org.wso2.carbon.remote.user.store.agent.manager.common.UserStoreManagerBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import javax.sql.DataSource;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.remote.user.store.agent.util.DatabaseUtils.createUserStoreDataSource;
+
+/**
+ * REST endpoint for authentication.
+ * This will be available at https://localhost:8888/wso2agent/authenticate
+ */
+@Path(CommonConstants.APPLICATION_CONTEXT_PATH)
+public class UserManagement {
+
+    private static Logger log = LoggerFactory.getLogger(UserManagement.class);
+    private static Properties connectionProperties = new Properties();
+    private DataSource dataSource;
+
+    public UserManagement(Properties connectionProperties) {
+
+        this.dataSource = createUserStoreDataSource(connectionProperties);
+        log.debug("Successfully created the datasource");
```

Comment:
Add the condition to check whether debug logs are enabled or not? and check this throughout the code.

Reply:
Fixed in 09486ce0c6187db78c1b97e4eaefd890c511141b.